modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -3385,7 +3385,13 @@ def sort_columns_by_row_values(self, rows, ascending=True, **kwargs):
     # Cat operations
     def cat_codes(self):
         def func(df) -> np.ndarray:
-            ser = df.iloc[:, 0]
+            # `df` is supposed to be consisted of multiple partitions,
+            # which should be concatenated before applying a function.
+            # `pd.concat` doesn't preserve categorical dtype
+            # if the dfs have categorical columns
+            # so we intentionaly restore the right dtype.
+            # TODO: revert the change when https://github.com/pandas-dev/pandas/issues/51362 is fixed.
```

Comment:
we've already raised a similar issue in modin, we should probably link it here as well: #2513

Reply:
This PR doesn't fix #2513, but only series.cat.* methods. So I think we can close #2513 when pandas has a fix on their side unless we get a P0 request to fix #2513 by that time.